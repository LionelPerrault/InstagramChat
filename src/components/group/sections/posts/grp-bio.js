import React from 'react'
import ToTags from '../../../hashtag/toTags/toTags'
import { Me } from '../../../../utils/utils'
import { connect } from 'react-redux'
import AppLink from '../../../others/link/link'

const GroupBio = ({ gd }) => {
  let { group_id, bio, admin } = gd

  return (
    <div className="grp_bio">
      <div className="grp_bio_h">
        <span>Group's bio</span>
      </div>

      <div className="grp_bio_main">
        {bio ? (
          <span>
            <ToTags str={bio} />
          </span>
        ) : (
          <span className="no_grp_bio">Group has no bio</span>
        )}
        {Me(admin) && (
          <AppLink
            url={`/group/${group_id}/edit`}
            className="sec_btn grp_ns"
            label="Not satisfied"
          />
        )}
      </div>
    </div>
  )
}

const mapStateToProps = state => ({ gd: state.Group.group_details })

export default connect(mapStateToProps)(GroupBio)
