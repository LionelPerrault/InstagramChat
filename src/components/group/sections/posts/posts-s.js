import React, { Component } from 'react'
import { connect } from 'react-redux'
import { FadeIn } from 'animate-components'
import PostItTeaser from '../../../post/post-it/post-it-teaser'
import Post from '../../../post/post/post'
import { getMutualAndNewestMembers } from '../../../../actions/group'
import MutualMembers from '../../mutual-members/mutual-members'
import NewestMembers from '../../newest-members/newest-members'
import CreateGroup from '../../create-group/create-group'
import GroupHashtags from '../../../hashtag/group-hashtags'
import GroupBio from './grp-bio'
import MapPosts from '../../../post/map-posts/map-posts'
import SectionsEnd from '../../../others/sections-end'
import AppLink from '../../../others/link/link'

class GroupPosts extends Component {
  componentDidMount = () => {
    let { dispatch, grp_id } = this.props
    dispatch(getMutualAndNewestMembers(grp_id))
  }

  render() {
    let {
        gd: { group_id, name },
        posts,
        joined,
      } = this.props,
      len = posts.length,
      map_posts = posts.map(p => (
        <Post key={p.post_id} {...p} when="groupPosts" />
      ))

    return (
      <div>
        <FadeIn duration="300ms">
          <div className="senapati pro_senapati">
            <div className="srajkumar">
              <GroupBio />
              <NewestMembers group={group_id} />
              <MutualMembers group={group_id} />
              <GroupHashtags group={group_id} />
              <CreateGroup />

              <div className="recomm_teaser">
                <span>Explore more groups from all around Instagram.</span>
                <AppLink
                  url="/explore/explore-groups"
                  className="sec_btn"
                  label="Explore"
                />
              </div>
            </div>

            <div className="prajkumar">
              {joined && <PostItTeaser type="group" group={group_id} />}
              <MapPosts
                posts={map_posts}
                nothingMssg={`${name} group has no posts!!`}
              />
            </div>
          </div>

          <SectionsEnd len={len} />
        </FadeIn>
      </div>
    )
  }
}

const mapStateToProps = store => ({
  gd: store.Group.group_details,
  joined: store.Group.joined,
  posts: store.Post.posts,
})

export default connect(mapStateToProps)(GroupPosts)
export { GroupPosts as PureGroupPosts }
