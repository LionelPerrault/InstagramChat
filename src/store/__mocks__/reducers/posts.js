export default [
  {
    post_id: 89,
    user: 24,
    username: 'takkar',
    firstname: 'iam_',
    surname: 'takkar',
    description: 'he @nobita, @doraemon #checkout, #dd #fgf',
    imgSrc: 'instagram_1518972814710.jpg',
    filter: 'filter-ashby',
    location: 'A-301, 90 Feet Road, Dharavi, Mumbai, Maharashtra 400017, India',
    type: 'user',
    group_id: 0,
    post_time: '1480114098767',
    tags_count: 1,
    likes_count: 3,
    shares_count: 3,
    comments_count: 2,
    group_name: '',
  },
  {
    post_id: 88,
    user: 24,
    username: 'takkar',
    firstname: 'iam_',
    surname: 'takkar',
    description: 'Hello @ghalib @takkar #checkout',
    imgSrc: 'instagram_1518945386167.jpg',
    filter: 'filter-normal',
    location: '',
    type: 'user',
    group_id: 0,
    post_time: '1480114098767',
    tags_count: 0,
    likes_count: 0,
    shares_count: 1,
    comments_count: 3,
    group_name: '',
  },
]
