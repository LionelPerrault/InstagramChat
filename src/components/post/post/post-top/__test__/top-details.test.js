import React from 'react'
import { create } from 'react-test-renderer'
import PostTopDetails from '../top-details'
import posts from '../../../../../store/__mocks__/reducers/posts'

describe('PostTopDetails Component', () => {
  it('should match snapshot', () => {
    const tree = create(<PostTopDetails details={{ ...posts[0] }} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot and show group name when type == group', () => {
    const tree = create(
      <PostTopDetails
        details={{
          ...posts[0],
          type: 'group',
          group_id: 11,
          group_name: 'cool group',
        }}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot and show first & second name of user if no location', () => {
    const tree = create(
      <PostTopDetails
        details={{
          ...posts[0],
          location: '',
        }}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
