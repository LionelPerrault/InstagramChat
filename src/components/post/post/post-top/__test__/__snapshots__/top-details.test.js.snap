// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTopDetails Component should match snapshot 1`] = `
Array [
  <div
    className="p_i_img"
  >
    <img
      src="/users/24/avatar.jpg"
    />
  </div>,
  <div
    className="p_i_1"
    style={
      Object {
        "top": "inherit",
      }
    }
  >
    <Link
      title="takkar"
      to="/profile/takkar"
    >
      takkar
    </Link>
    <span
      className="p_i_1_title"
      title="A-301, 90 Feet Road, Dharavi, Mumbai, Maharashtra 400017, India"
    >
      A-301, 90 Feet Road, Dharavi, Mum..
    </span>
  </div>,
]
`;

exports[`PostTopDetails Component should match snapshot and show first & second name of user if no location 1`] = `
Array [
  <div
    className="p_i_img"
  >
    <img
      src="/users/24/avatar.jpg"
    />
  </div>,
  <div
    className="p_i_1"
    style={
      Object {
        "top": "inherit",
      }
    }
  >
    <Link
      title="takkar"
      to="/profile/takkar"
    >
      takkar
    </Link>
    <span
      className="p_i_1_title"
      title={null}
    >
      iam_ takkar
    </span>
  </div>,
]
`;

exports[`PostTopDetails Component should match snapshot and show group name when type == group 1`] = `
Array [
  <div
    className="p_i_img"
  >
    <img
      src="/users/24/avatar.jpg"
    />
  </div>,
  <div
    className="p_i_1"
    style={
      Object {
        "top": -8,
      }
    }
  >
    <Link
      title="takkar"
      to="/profile/takkar"
    >
      takkar
    </Link>
    <div
      className="its_grp_post"
    >
      <span
        className="to_grp_arrow"
      >
        <i
          className="material-icons"
        >
          arrow_drop_up
        </i>
      </span>
      <Link
        className="to_grp_name"
        to="/group/11"
      >
        cool group
      </Link>
    </div>
    <span
      className="p_i_1_title"
      title="A-301, 90 Feet Road, Dharavi, Mumbai, Maharashtra 400017, India"
    >
      A-301, 90 Feet Road, Dharavi, Mum..
    </span>
  </div>,
]
`;
